### PR TITLE
feat: Serialize directly from refract to v8 objects

### PR DESCRIPTION
**Pull request by w-vi**
Initial attempt on using refract directly.

**DO NOT MERGE** as the serialization differs from what drafter currently does.

### CODE REVIEW

File: src/protagonist.cc [hunk 1]
```diff
@@ -5,18 +5,16 @@ using namespace protagonist;
...
     exports->Set(Nan::New<String>("parse").ToLocalChecked(), Nan::New<FunctionTemplate>(Parse)->GetFunction());
+
+    // parseSync function
     exports->Set(Nan::New<String>("parseSync").ToLocalChecked(), Nan::New<FunctionTemplate>(ParseSync)->GetFunction());
 
-    // Validate function
-    exports->Set(Nan::New<String>("validate").ToLocalChecked(), Nan::New<FunctionTemplate>(Validate)->GetFunction());
+    // validate function
+    exports->Set(Nan::New<String>("validate").ToLocalChecked(), Nan::New<FunctionTemplate>(Validate)->GetFunction())
+        ;
```
Comment (pksunkara) [inline src/protagonist.cc]: Bad style.

**Comment by goganchic** [comment]
```
FORMAT: 1A

# Group Test

### Test [GET /foo]

+ Request (application/json)
    + Attributes
        + One of
            + a
            + b
```

parsing such document gives me segmentation fault (in protagonist only, in plain drafter master everything is ok)

If I changes document to this:
```
FORMAT: 1A

# Group Test

### Test [GET /foo]

+ Request (application/json)
        + One of
            + a
            + b
```

everything is fine. Is it a bug or my document is buggy?

**Comment by goganchic** [comment]
https://github.com/apiaryio/protagonist/blob/9a599816adbf811e1ec78b3e4d90ddfbd8c57c13/src/refract2v8.cc#L346 here `m == null`

**Comment by w-vi** [comment]
@goganchic It very likely is a bug it has not been thoroughly tested yet because we need to figure out the serialization of refract in general first. But good catch, thanks a lot.

**Comment by kylef** [comment]
Completed in #189

**Pull request closed without merging**